core/internal/testutils/evmtest/evmtest.go

```diff
@@ -36,7 +36,7 @@ import (
 )
 
 type TestChainOpts struct {
-	Client         interface{} // evmclient.Client or func() evmclient.Client
+	Client         evmclient.Client
```

Comment:
@reductionista I ran in to type trouble with this field. What do you think about this alternative approach of splitting the helper so that we can modify the opts after they are converted from test-opts?